### PR TITLE
Kamu 165 set watermark  add event from UI

### PR DESCRIPTION
**Pull request by dmitriy-borzenko**

### CODE REVIEW

File: src/app/dataset-view/additional-components/overview-component/components/edit-watermark-modal/edit-watermark-modal.component.ts [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    ],
+})
+export class EditWatermarkModalComponent
+    extends BaseComponent
+    implements OnInit
+{
+    @Input() public currentWatermark: MaybeNull<string>;
+    @Input() public datasetBasics?: DatasetBasicsFragment;
+    public date: Date | string;
+    public tooltip = "Description of the watermark.";
```
Comment (zaychenko-sergei) [inline src/app/dataset-view/additional-components/overview-component/components/edit-watermark-modal/edit-watermark-modal.component.ts:32]: readonly, uppercase

File: src/app/dataset-view/additional-components/overview-component/overview-component.html [hunk 2]
```diff
@@ -286,18 +286,51 @@ <h2 class="dataset-info--subtitle">
                     </div>
                     <div
                         class="dataset-info--block"
-                        *ngIf="currentState && currentState.overview.metadata.currentWatermark"
+                        *ngIf="currentState.overview.kind==='ROOT'"
```
Comment (zaychenko-sergei) [inline src/app/dataset-view/additional-components/overview-component/overview-component.html:289]: I am not sure this condition is valid. Why do you think this should be disallowed for derived datasets?

**Review by zaychenko-sergei — CHANGES_REQUESTED** [review]
Perhaps we also need this edit link on Metadata tab:

image

File: src/app/dataset-view/additional-components/overview-component/components/edit-watermark-modal/edit-watermark-modal.component.ts (hunk 1 above)
Comment (dmitriy-borzenko) [reply src/app/dataset-view/additional-components/overview-component/components/edit-watermark-modal/edit-watermark-modal.component.ts:32]: Deleted this property

File: src/app/dataset-view/additional-components/overview-component/overview-component.ts [hunk 3]
```diff
@@ -112,6 +113,19 @@ export class OverviewComponent extends BaseComponent implements OnInit {
         ).datasetBasics = this.datasetBasics;
     }
 
+    public openWatermarkModal(): void {
+        const modalRef: NgbModalRef = this.modalService.open(
+            EditWatermarkModalComponent,
+        );
+        (
+            modalRef.componentInstance as EditWatermarkModalComponent
+        ).currentWatermark = this.currentState?.overview.metadata
```
Comment (zaychenko-sergei) [inline src/app/dataset-view/additional-components/overview-component/overview-component.ts:122]: Maybe do this cast just once?

**Comment by zaychenko-sergei** [comment]
Don't forget Changelog entry

**Pull request merged**